src/Cookies.php

```diff
@@ -316,7 +321,11 @@ protected function _matchPath($domains, $path)
         $ret = [];
 
         foreach ($domains as $dom => $pathsArray) {
-            foreach (array_keys($pathsArray) as $cpath) {
+            $keys = array_keys($pathsArray);
+            foreach ($keys as $cpath) {
+                if (! \is_string($cpath)) {
+                    throw new InvalidArgumentException('Domains is not a valid array');
+                }
```

Comment:
I read the resolved conversation above, and have a few comments here:

- Why can't `$domains` be documented as `array<string, array<string, mixed>>`?
- Same note as previously about importing functions.
- This is new behavior and requires a corresponding test.

Reply:
> Why can't `$domains` be documented as `array<string, array<string, mixed>>`?

Done. You're right, when we'll have type hint `SetCookie::matchCookiePath()` parameter will just accept a string.